File: src/main/java/akko/ddbot/sql/SQLFun.java
package akko.ddbot.sql;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.SQLException;

public class SQLFun {
    public Boolean execute(String databaseName, String sql)
    {
        try {
            Class.forName("org.sqlite.JDBC");
            Connection sqliteC = DriverManager.getConnection("jdbc:sqlite:db/" + databaseName +".db");
            Boolean res = sqliteC.prepareStatement(sql).execute();
            sqliteC.close();
            return res;
        }
        catch (ClassNotFoundException | SQLException e)
        {
            System.out.println(e.getMessage());
            e.printStackTrace();
        }
        return false;
    }

    public TwoTuple<ResultSet,Connection> executeQuery(String databaseName, String sql)
    {
        try {
            Class.forName("org.sqlite.JDBC");
            Connection sqliteC = DriverManager.getConnection("jdbc:sqlite:db/" + databaseName +".db");
            ResultSet res = sqliteC.prepareStatement(sql).executeQuery();
            return new TwoTuple<ResultSet, Connection>(res,sqliteC);
        }
        catch (ClassNotFoundException | SQLException e)
        {
            System.out.println(e.getMessage());
            e.printStackTrace();
        }
        return null;
    }

    public Connection connection(String databaseName)
    {
        try
        {
            Class.forName("org.sqlite.JDBC");
            return DriverManager.getConnection("jdbc:sqlite:db/" + databaseName + ".db");
        }
        catch (ClassNotFoundException | SQLException e)
        {
            System.out.println(e.getMessage());
            e.printStackTrace();
        }
        return null;
    }
}

File: src/main/java/akko/ddbot/sql/SQLFun.kt
package akko.ddbot.sql

import akko.ddbot.BotMainActivity
import akko.ddbot.Init
import java.sql.*

class SQLFun {
    fun execute(sql: String): Boolean
    {
        try {
            Class.forName("org.postgresql.Driver")
            val sqliteC = DriverManager.getConnection("jdbc:postgresql://${Init.POSTGRE_URL}/${Init.POSTGRE_DATABASE}", Init.POSTGRE_USER, Init.POSTGRE_PASSWD)
            val res = sqliteC.prepareStatement(sql).execute()
            sqliteC.close()
            return res
        } catch (e: ClassNotFoundException) {
            BotMainActivity.SQLLogger!!.debug(e.message)
        } catch (e: SQLException) {
            BotMainActivity.SQLLogger!!.debug(e.message)
        }
        return false
    }
    fun executeQuery(sql: String?): Pair<ResultSet, Connection>? {
        try {
            Class.forName("org.postgresql.Driver")
            val sqliteC = DriverManager.getConnection("jdbc:postgresql://${Init.POSTGRE_URL}/${Init.POSTGRE_DATABASE}",Init.POSTGRE_USER,Init.POSTGRE_PASSWD)
            val res = sqliteC.prepareStatement(sql).executeQuery()
            res.next()
            return Pair<ResultSet, Connection>(res, sqliteC)
        } catch (e: ClassNotFoundException) {
            BotMainActivity.SQLLogger!!.debug(e.message)
        } catch (e: SQLException) {
            BotMainActivity.SQLLogger!!.debug(e.message)
        }
        return null
    }
    fun connection(): Connection? {
        try {
            Class.forName("org.postgresql.Driver")
            return DriverManager.getConnection("jdbc:postgresql://${Init.POSTGRE_URL}/${Init.POSTGRE_DATABASE}",Init.POSTGRE_USER,Init.POSTGRE_PASSWD)
        } catch (e: ClassNotFoundException) {
            BotMainActivity.SQLLogger!!.debug(e.message)
        } catch (e: SQLException) {
            BotMainActivity.SQLLogger!!.debug(e.message)
        }
        return null
    }
    fun PreparedStatement(sql: String): Pair<PreparedStatement,Connection>
    {
        Class.forName("org.postgresql.Driver")
        val conn = DriverManager.getConnection("jdbc:postgresql://${Init.POSTGRE_URL}/${Init.POSTGRE_DATABASE}",Init.POSTGRE_USER,Init.POSTGRE_PASSWD)
        return Pair<PreparedStatement,Connection>(conn.prepareStatement(sql),conn)
    }
}